fix setting unix socket authentication on mariadb >= 10.4

Vringe opened the pull request:
### What does this PR do?

Quote from the [MariaDB Docs](https://mariadb.com/kb/en/mysqluser-table/):
> In [MariaDB 10.4](https://mariadb.com/kb/en/what-is-mariadb-104/) and later, the [mysql.global_priv table](https://mariadb.com/kb/en/mysqlglobal_priv-table/) has replaced the mysql.user table, and mysql.user should be considered obsolete. It is now a [view](https://mariadb.com/kb/en/views/) into mysql.global_priv created for compatibility with older applications and monitoring scripts.

Example state to reproduce (A user must already exist with an other authentication plugin than unix socket):
```
sql_root_user:
  mysql_user.present:
    - name: root
    - host: 'localhost'
    - allow_passwordless: True
    - unix_socket: True
```

### Previous Behavior
Salt still uses `UPDATE mysql.user` when changing a users authentication plugin to unix_socket. This leads to a corrupted View:
```
          ID: sql_root_user
    Function: mysql_user.present
        Name: root
      Result: False
     Comment: Failed to clear password for user root@localhost (MySQL Error 1356: View 'mysql.user' references invalid table(s) or column(s) or function(s) or definer/invoker of view lack rights to use them)
     Started: 15:36:08.057118
    Duration: 18.406 ms
     Changes:
```

### New Behavior
User is altered correctly on older and newer MariaDB versions.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**
<!-- Please review the [test documentation](https://docs.saltproject.io/en/master/topics/tutorials/writing_tests.html) for details on how to implement tests into Salt's test suite. -->
- [ ] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
No

Commit:
use ALTER USER on mariadb >= 10.4 when switching to unix socket authentication

Ch3LL reviewed (changes requested):
This will require a changelog and test coverage. We do have tests for version 10.4 so we would have caught this if the correct test was added.  A good place to add the test is here: tests/pytests/integration/modules/test_mysql.py

Ch3LL commented:
@Vringe would you be able to loop back to this PR to add changelog and test coverage?

Vringe commented:
@Ch3LL Unfortunately, I do not have the time at the moment. I tried to setup the dev environment a while ago to write and run the test but I could not get it to run properly.

danel1 commented:
@garethgreenaway @pprkut Hey guys :) i saw you both did some other work for MariaDB 10.5+ support in other issues / pr's. Any chance you can help this pr out?

Vringe commented:
@cebe I saw you did some MariaDB related work for salt 3006. Unfortunately this issue still persists. Could you maybe help me out with this?

twangboy reviewed (changes requested):
Please create a changelog and a test for this

Commit:
Add changelog and regression test for PR #62169

Adds the missing changelog fragment and a unit test pinning the new
MariaDB >= 10.4 ``ALTER USER ... IDENTIFIED VIA unix_socket`` code path
in ``_mariadb_user_chpass``. The test also verifies the legacy
``UPDATE mysql.user`` path is still used on MariaDB < 10.4.
